src/modules/keyboardmanager/KeyboardManagerEngineLibrary/KeyboardEventHandlers.cpp

```diff
@@ -822,6 +930,708 @@ namespace KeyboardEventHandlers
+                    currentDirPtr = nullptr;
+                    toast(title, message);
+                    return;
+                }
+            }
+
+            DWORD processId = 0;
+            HANDLE newProcessHandle;
+
+            if (shortcut.elevationLevel == Shortcut::ElevationLevel::Elevated)
+            {
+                newProcessHandle = run_elevated(fullExpandedFilePath, shortcut.runProgramArgs, currentDirPtr, (shortcut.startWindowType == Shortcut::StartWindowType::Normal));
+                processId = GetProcessId(newProcessHandle);
+            }
+            else if (shortcut.elevationLevel == Shortcut::ElevationLevel::NonElevated)
+            {
+                run_non_elevated(fullExpandedFilePath, shortcut.runProgramArgs, &processId, currentDirPtr, (shortcut.startWindowType == Shortcut::StartWindowType::Normal));
+            }
+            else if (shortcut.elevationLevel == Shortcut::ElevationLevel::DifferentUser)
+            {
+                newProcessHandle = run_as_different_user(fullExpandedFilePath, shortcut.runProgramArgs, currentDirPtr, (shortcut.startWindowType == Shortcut::StartWindowType::Normal));
+                processId = GetProcessId(newProcessHandle);
+            }
+
+            if (processId == 0)
+            {
+                std::wstring title = fmt::format(L"Error starting {}", fileNamePart);
+                std::wstring message = fmt::format(L"The application might not have started.");
+                toast(title, message);
+                return;
+            }
+
+            if (shortcut.startWindowType == Shortcut::StartWindowType::Hidden)
+            {
+                HideProgram(processId, fileNamePart, 0);
+            }
+            //ShowProgram(processId, fileNamePart, true, false, (shortcut.startWindowType == Shortcut::StartWindowType::Hidden), 0);
+        }
+        return;
+    }
+
+    void CloseAndTerminateProcessByName(const std::wstring& fileNamePart)
+    {
+        CloseProcessByName(fileNamePart);
+        Sleep(1000);
+        TerminateProcessesByName(fileNamePart);
+    }
+
+    void CloseProcessByName(const std::wstring& fileNamePart)
+    {
+        auto processIds = GetProcessesIdByName(fileNamePart);
+
+        if (processIds.size() == 0)
+        {
+            Logger::trace(L"ChordKeyboardHandler:{}, Nothing To WM_CLOSE", fileNamePart);
+            return;
+        }
+
+        if (false)
+        {
+            auto retryCount = 10;
+            while (processIds.size() > 0 && retryCount-- > 0)
+            {
+                for (DWORD pid : processIds)
+                {
+                    //Logger::trace(L"ChordKeyboardHandler:{}, WM_CLOSE ({}) -> pid:{}", fileNamePart, retryCount, pid);
+                    HWND hwnd = FindMainWindow(pid, false);
+                    SendMessage(hwnd, WM_CLOSE, 0, 0);
+                }
+
+                Sleep(500);
```

Comment:
Another Sleep here? Will this run on the keyboard handler? That's a no-no if that's the case. Thinking this needs to  be on another thread. Heavy operations shouldn't be on the keyboard handlers as that will cause input lag and if it takes too long Windows will even silently remove keyboard manager from receiving key presses in the low-level keyboard hook,.

Reply:
The "Sleep(500)" there is in an `if(false)` so I will remove that. 

The part that actually does run IS in a thread already. There was a `Sleep(100)` outside the thread code context, and I removed that also and it seems to be working fine still.